emily/handler/src/common/error.rs

```diff
@@ -128,21 +97,130 @@ pub enum Error {
     InconsistentState(Inconsistency),
 
     /// API is reorganizing.
-    #[error("Api is reorganizing around new chain tip {0:?}")]
+    #[error("the API is reorganizing around new chain tip {0:?}")]
     Reorganizing(Chainstate),
 
     /// An entry update version conflict in a resource update resulted
     /// in an update not being performed.
-    #[error("Version conflict")]
-    VersionConflict,
-
-    /// Bad request
-    #[error("Bad request {0}")]
-    BadRequest(String),
+    #[error("there was a conflict when attempting to update the database; {0}")]
+    VersionConflict(#[source] Box<ConditionalCheckFailedException>),
 
     /// Deserialization error
     #[error("Deserialization error: {0}")]
     Deserialization(String),
+
+    /// This happens if the deposit entry that was stored in the database
+    /// was invalid, or if the deposit entry that we are creating to store
+    /// in the database is invalid.
+    #[error("deposit entry failed validation; {0}; ID: {1}")]
+    DepositEntry(&'static str, DepositEntryKey),
+
+    /// This happens when there is a mismatch in the outpoint of the new
+    /// deposit event and the fetched deposit entry. Seeing this is
+    /// probably due to a programming error.
+    #[error("mismatch when updateing deposit request; existing: {0}; update: {1}")]
+    DepositUpdate(DepositEntryKey, DepositEntryKey),
+
+    /// This happens if the withdrawal entry that was stored in the database
+    /// was invalid, or if the withdrawal entry that we are creating to store
+    /// in the database is invalid.
+    #[error("withdrawal entry failed validation; {0}; ID: {1}")]
+    WithdrawalEntry(&'static str, WithdrawalEntryKey),
+
+    /// This happens when there is a mismatch in the request ID of the new
+    /// withdrawal event and the fetched withdrawal entry. Seeing this is
+    /// probably due to a programming error.
+    #[error("mismatch when updateing withdrawal request; existing: {0}; update: {1}")]
```

Comment:
```suggestion
    #[error("mismatch when updating withdrawal request; existing: {0}; update: {1}")]
```

Reply:
Fixed in https://github.com/stacks-sbtc/sbtc/pull/1627/commits/00dd15b8f7d4e8a807fdc80b032296d7f8c2b9ed